test/active_record_test.rb

```diff
@@ -12,125 +12,131 @@ def teardown
     super
   end
 
-  test "store_translations does not allow ambiguous keys (1)" do
-    I18n::Backend::ActiveRecord::Translation.delete_all
-    I18n.backend.store_translations(:en, :foo => 'foo')
-    I18n.backend.store_translations(:en, :foo => { :bar => 'bar' })
-    I18n.backend.store_translations(:en, :foo => { :baz => 'baz' })
+  [false, true].each do |cache_translations|
+    I18n::Backend::ActiveRecord.config.cache_translations = cache_translations
```

Comment:
I don't think this is going to work.

From my understanding the `test` call defines a method, but when those methods are going to be executed the current state of the config(in our case `true`) will be used.

This can easily tested by adding something like
```ruby
puts "cache_translations: #{I18n::Backend::ActiveRecord.config.cache_translations}"
```
inside each test method and I am almost sure it will be always `true`

So instead, we should add the config call inside every `test` method
```ruby
test "store_translations does not allow ambiguous keys (1), Cache Translations: #{cache_translations}" do
  I18n::Backend::ActiveRecord.config.cache_translations = cache_translations
  I18n::Backend::ActiveRecord::Translation.delete_all
  # ...
end
```